src/init.cpp

```diff
@@ -591,7 +591,7 @@ void SetupServerArgs(ArgsManager& argsman)
 
 
     argsman.AddArg("-blockmaxweight=<n>", strprintf("Set maximum BIP141 block weight (default: %d)", DEFAULT_BLOCK_MAX_WEIGHT), ArgsManager::ALLOW_ANY, OptionsCategory::BLOCK_CREATION);
-    argsman.AddArg("-blockmintxfee=<amt>", strprintf("Set lowest fee rate (in %s/kvB) for transactions to be included in block creation. (default: %s)", CURRENCY_UNIT, FormatMoney(DEFAULT_BLOCK_MIN_TX_FEE)), ArgsManager::ALLOW_ANY, OptionsCategory::BLOCK_CREATION);
+    argsman.AddArg("-blockmintxfee=<amt>", strprintf("(DEPRECATED) Set lowest fee rate (in %s/kvB) for transactions to be included in block creation. (default: %s)", CURRENCY_UNIT, FormatMoney(DEFAULT_BLOCK_MIN_TX_FEE)), ArgsManager::ALLOW_ANY, OptionsCategory::BLOCK_CREATION);
```

Comment:
Should say "UNSUPPORTED", not "DEPRECATED" ?

Weak NACK on removing this option -- in [inq#23](https://github.com/bitcoin-inquisition/bitcoin/pull/23#discussion_r1139697940) we've got a use case for accepting txs into the mempool that we don't intend to mine (until they are CPFP'ed).

Reply:
(Would support changing the default to 0sat/vb though)

Reply:
From the ephemeral anchors review club:

> `<glozow> ok hm. i kinda need to get rid of blockmintxfee for pakcage relay tho`

Can you elaborate on that? You can just drop the blockminfee commit from this PR, and it seems fine, no?

The only result of setting `-blockminfee` is that if its value is higher than `-minrelaytxfee` you'll potentially never empty your own memory even if you're the only miner and no new transactions are being created/relayed. That's undesirable as default behaviour, but seems okay if you deliberately choose it. (And the mempool will empty eventually anyway, as a result of expiry)

Reply:
> From the ephemeral anchors review club:

>>    `<glozow> ok hm. i kinda need to get rid of blockmintxfee for pakcage relay tho`

> Can you elaborate on that?

My thinking there was that we needed to address this DoS issue in order to proceed with package relay, and this PR seemed like the most likely approach (hence the "kinda"). Now, maybe #26933 seems more likely?

>  You can just drop the blockminfee commit from this PR, and it seems fine, no?
> The only result of setting -blockminfee is that if its value is higher than -minrelaytxfee you'll potentially never empty your own memory even if you're the only miner and no new transactions are being created/relayed.

I don't think the problem is only when blockmintxfee > minrelaytxfee. With this PR, if both are 1sat/vB, you can have (all transactions same size):
1. parent (paying 1 satoshi in fees) + fee-bumping child accepted to mempool as a package
2. attach 2 more children to the parent, each 1.5sat/vB
3. replace the child with something not bumping the parent. The parent is not removed because its descendant score > 1sat/vB
4. parent is also never selected in BlockAssembler because the 2 children's ancestor scores are < 1sat/vB.

This works even if we have a significantly lower blockmintxfee (e.g. 0.6sat/vB) while minrelaytxfee = 1sat/vB. We can still get to an equivalent state, just with more children. N children with feerate (N+1)/N keeps the descendant score above 1sat/vB while the ancestor scores are lower than blockmintxfee.

The main idea for this PR is to mine everything in mempool which requires blockmintxfee=0. Perhaps a middle ground is to set the default to 0?

Reply:
Yeah, fair point -- the claim should be "setting -blockminfee higher than zero may result in txs remaining in your mempool that will never be mined"; but as long as that's not the default behaviour, that doesn't seem particularly problematic? (TBH, I'd rather be able to pass blockminfee/blockmaxweight as getblocktemplate parameters)

Sounds like this should be closed though?

Reply:
Resolving - restored support for -blockmintxfee (now it's actually just "deprecated"), default is 0, and added this warning when it is set.